src/main/java/org/variantsync/diffdetective/variationtree/VariationNode.java

```diff
@@ -0,0 +1,583 @@
+
+    /**
+     * Returns {@code true} iff this node has no parent.
+     *
+     * @see getParent
+     */
+    public boolean isRoot() {
+        return getParent() == null;
+    }
+
+    /**
+     * Returns {@code true} iff this node has no children.
+     *
+     * @see getChildren
+     */
+    public boolean isLeaf() {
+        return getChildren().isEmpty();
+    }
+
+    /**
+     * Returns the number of child nodes.
+     *
+     * <p>Note: This is O(n) for {@link DiffNode.Projection}.
+     *
+     * @see getChildCount
+     */
+    public int getChildCount() {
+        return getChildren().size();
+    }
+
+    /**
+     * Computes the length of the path from the root node to this node.
+     */
+    public int getDepth() {
+        if (isRoot()) {
+            return 0;
+        }
+
+        return getParent().getDepth() + 1;
+    }
+
+    /**
+     * Computes the number of {@link isAnnotation annotations} of the path from the root node to
+     * this node.
+     */
+    public int getAnnotationDepth() {
+        if (isRoot()) {
+            return 0;
+        }
+
+        if (isIf()) {
+            return getParent().getAnnotationDepth() + 1;
+        }
+
+        return getParent().getAnnotationDepth();
+    }
+
+    /**
+     * Returns the first {@code if} node in the path from this node upwards to the root.
+     */
+    public VariationNode<T> getIfNode() {
+        if (isIf()) {
+            return this;
+        }
+        if (isRoot()) {
+            return null;
+        }
+        return getParent().getIfNode();
+    }
+
+    /**
+     * Returns {@code true} iff this node is the parent of the given node.
+     *
+     * @see getChildren
+     */
+    public boolean isChild(VariationNode<T> child) {
+        return child.getParent().isSameAs(child);
+    }
+
+    /**
+     * Returns the index of the given child in the list of children of this node.
+     * Returns -1 if the given node is not a child of this node.
+     *
+     * <p>Warning: If this is a {@link DiffNode.Projection}, then the returned index may be
+     * different to the index returned by {@link DiffNode#indexOfChild}.
+     *
+     * @see getChildren
+     */
+    public int indexOfChild(final VariationNode<T> child) {
+        return getChildren().indexOf(child);
+    }
+
+    /**
+     * The same as {@link addChild(VariationNode<T>, int)} but puts the node at the end of the
+     * children list instead of inserting it at a specific index.
+     *
+     * @see getChildren
+     */
+    public abstract void addChild(final VariationNode<T> child);
+
+    /**
+     * Adds a child before the given index to the children list of this node and sets its parent to
+     * this node.
+     *
+     * <p>When calling {@link indexOfChild} with {@code child} the returned index will be
+     * {@code index} as long as the children list isn't modified.
+     *
+     * @see addChildren
+     */
+    public abstract void addChild(final VariationNode<T> child, int index);
```

Comment:
insertChild?

Reply:
Oh yes. inconsistent with `List` or `DiffNode`? Well I agree being consistent with `DiffNode` is probably better.

Reply:
Yes I would also prefer being consistent with DiffNode.